viewflow/lock.py

```
@@ -77,3 +95,29 @@ def lock(flow_cls, process_pk):
             cache.delete(key)
 
     return lock
+
+
+def redis_lock(flow, attempts=5, expires=120):
+    """Task lock based redis locks implemented in ``django-redis``."""
+    @contextmanager
+    def lock(flow_cls, process_pk):
+        key = 'django-viewflow-lock-{}/{}'.format(flow_cls._meta.namespace, process_pk)
+
+        for i in range(attempts):
+            lock = cache.lock(key, timeout=expires)
```

I will merge it as is now, since we are using it this way, but eventually we will need to throw an exception